### ISSUE TITLE
GPU training on python 3.10

### ISSUE THREAD

**Issue by B33low** [nicknochnack/TFODCourse#86]
After hours of unsuccessful research for making the gpu work for python 3.10, I found the corresponding CUDA and cuDNN working:

**Python 3.10
Tensorflow 2.8
CUDA 11.2
cuDNN 8.2.1 for CUDA 11.x**

I hope this will help some people

**Comment by matthewtesimale** [comment]
Did you downgrade from tensorflow 2.9.1 or was that just the latest version at the time?

Also, did you do any other folder/file arranging outside of what he did in the video?

Thanks for sharing!!! :)

**Comment by B33low** [comment]
At the time this was the latest version if I remember correctly yes

**Comment by Limondeeplearning** [comment]
will this fix this error:

ERROR: Could not find a version that satisfies the requirement tensorflow==2.4.1 (from versions: 2.8.0rc1, 2.8.0, 2.8.1, 2.8.2, 2.8.3, 2.9.0rc0, 2.9.0rc1, 2.9.0rc2, 2.9.0, 2.9.1, 2.9.2, 2.10.0rc0, 2.10.0rc1, 2.10.0rc2, 2.10.0rc3, 2.10.0)
ERROR: No matching distribution found for tensorflow==2.4.1

**Comment by B33low** [comment]
Did you manage to solve this problem ? This would be interesting to have some review about your situation

**Comment by Limondeeplearning** [comment]
I managed to make the entire program work but using cpu  cause I couldn't find the right pytorch version for my cuda version which is 11.2, but  I was able to train using my gpu...

Ps. I'm actually doing the ANPR tutorial but you know nick leveraged this code/system for that project...

**Comment by Limondeeplearning** [comment]
I upgraded from python 3.8 to 3.10 then from tf 2.4.1 to 2.10.0

here is my full list (PS. again ...I did the ANPR video so idk if there's additonal packages here thats not on the original TFOD course.. well except the OCR and Pytorch I think,,haha) hope this will help people. I might actually post this on the ANPR repo for others to view... 
Package                       Version     Editable project location
----------------------------- ----------- -----------------------------------------------------------
absl-py                       1.3.0
asttokens                     2.1.0
astunparse                    1.6.3
backcall                      0.2.0
cachetools                    5.2.0
certifi                       2022.9.24
charset-normalizer            2.1.1
colorama                      0.4.6
contourpy                     1.0.6
cycler                        0.11.0
Cython                        0.29.32
debugpy                       1.6.3
decorator                     5.1.1
dill                          0.3.6
dm-tree                       0.1.7
easyocr                       1.6.2
entrypoints                   0.4
etils                         0.9.0
executing                     1.2.0
flatbuffers                   22.10.26
fonttools                     4.38.0
gast                          0.4.0
gin-config                    0.5.0
google-api-core               2.10.2
google-api-python-client      2.66.0
google-auth                   2.14.1
google-auth-httplib2          0.1.0
google-auth-oauthlib          0.4.6
google-pasta                  0.2.0
googleapis-common-protos      1.57.0
grpcio                        1.50.0
h5py                          3.7.0
httplib2                      0.21.0
idna                          3.4
imageio                       2.22.4
immutabledict                 2.2.3
importlib-resources           5.10.0
ipykernel                     6.17.1
ipython                       8.6.0
ipywidgets                    8.0.2
jedi                          0.18.1
joblib                        1.2.0
jupyter_client                7.4.7
jupyter_core                  5.0.0
jupyterlab-widgets            3.0.3
kaggle                        1.5.12
keras                         2.10.0
Keras-Preprocessing           1.1.2
kiwisolver                    1.4.4
libclang                      14.0.6
lvis                          0.5.3
lxml                          4.9.1
Markdown                      3.4.1
MarkupSafe                    2.1.1
matplotlib                    3.6.2
matplotlib-inline             0.1.6
nest-asyncio                  1.5.6
networkx                      2.8.8
ninja                         1.11.1
numpy                         1.23.4
oauth2client                  4.1.3
oauthlib                      3.2.2
object-detection              0.1
opencv-python                 4.6.0.66
opt-einsum                    3.3.0
packaging                     21.3
pandas                        1.5.1
parso                         0.8.3
pickleshare                   0.7.5
Pillow                        9.3.0
pip                           22.3.1
platformdirs                  2.5.4
portalocker                   2.6.0
promise                       2.3
prompt-toolkit                3.0.32
protobuf                      3.19.6
psutil                        5.9.4
pure-eval                     0.2.2
py-cpuinfo                    9.0.0
pyasn1                        0.4.8
pyasn1-modules                0.2.8
pyclipper                     1.3.0.post4
pycocotools                   2.0.6
Pygments                      2.13.0
pyparsing                     3.0.9
python-bidi                   0.4.2
python-dateutil               2.8.2
python-slugify                6.1.2
pytz                          2022.6
PyWavelets                    1.4.1
pywin32                       305
PyYAML                        5.4.1
pyzmq                         24.0.1
regex                         2022.10.31
requests                      2.28.1
requests-oauthlib             1.3.1
rsa                           4.9
sacrebleu                     2.2.0
scikit-image                  0.19.3
scikit-learn                  1.1.3
scipy                         1.9.3
sentencepiece                 0.1.97
seqeval                       1.2.2
setuptools                    65.5.1
Shapely                       1.8.5.post1
six                           1.16.0
slim                          0.1         c:\users\administrator\anpr\tensorflow\models\research\slim
stack-data                    0.6.1
tabulate                      0.9.0
tensorboard                   2.10.1
tensorboard-data-server       0.6.1
tensorboard-plugin-wit        1.8.1
tensorflow                    2.10.0
tensorflow-addons             0.18.0
tensorflow-datasets           4.7.0
tensorflow-estimator          2.10.0
tensorflow-gpu                2.10.0
tensorflow-hub                0.12.0
tensorflow-io                 0.27.0
tensorflow-io-gcs-filesystem  0.27.0
tensorflow-metadata           1.11.0
tensorflow-model-optimization 0.7.3
tensorflow-text               2.10.0
termcolor                     2.1.0
text-unidecode                1.3
tf-models-official            2.10.1
tf-slim                       1.1.0
threadpoolctl                 3.1.0
tifffile                      2022.10.10
toml                          0.10.2
torch                         1.13.0
torchaudio                    0.13.0
torchvision                   0.14.0
tornado                       6.2
tqdm                          4.64.1
traitlets                     5.5.0
typeguard                     2.13.3
typing_extensions             4.4.0
uritemplate                   4.1.1
urllib3                       1.26.12
wcwidth                       0.2.5
Werkzeug                      2.2.2
wget                          3.2
wheel                         0.38.4
widgetsnbextension            4.0.3
wrapt                         1.14.1
zipp                          3.10.0
